### PR TITLE
PR: Added shortcut and menu option to re-run last cell

### LINKED ISSUE #3959: Shortcut feature request: re-run the previous cell
like the button "re-run the previous code region" in Rstudio.
Thank you.

### PR DESCRIPTION
**Pull request by andfoy**
Fixes #3959

### CODE REVIEW

**Comment by ccordoba12** [comment]
@andfoy, this one also has conflicts with 3.x. Please update it so I can merge it :-)

File: spyder/config/main.py [hunk 1]
```diff
@@ -57,10 +57,13 @@
 # Run cell shortcuts
 if sys.platform == 'darwin':
     RUN_CELL_SHORTCUT = 'Meta+Return'
+    RE_RUN_LAST_CELL_SHORTCUT = 'Meta+Shift+R'
 else:
     RUN_CELL_SHORTCUT = 'Ctrl+Return'
+    RE_RUN_LAST_CELL_SHORTCUT = 'Ctrl+Shift+R'
```
Comment (ccordoba12) [inline spyder/config/main.py]: `Ctrl+Shift+...` is reserved to give focus to plugins. Please change this one to `Alt+Shift+R`

**Comment by ccordoba12** [comment]
Could you post a screenshot to see how the new button looks?

**Comment by ccordoba12** [comment]
The icon is a bit overloaded, so I don't think we should use it for now (it took us a lot of time to come up with good icons for the run cell actions, so don't worry about it).

So to finish this PR I think you need to:

1. Remove the icon you introduced.
2. Leave the re-run cell action in the Run menu and remove it from the Run toolbar.
3. Change its shortcut to `Alt+Enter`. Sorry for making you to change it again, but it seems more logical to me now to have all these shortcuts related to `Modifier+Enter`.

**Comment by andfoy** [comment]
@ccordoba12 This is ready to go

File: spyder/plugins/editor.py [hunk 2]
```diff
@@ -105,9 +105,12 @@ def setup_page(self):
         
         interface_group = QGroupBox(_("Interface"))
         newcb = self.create_checkbox
+        fpsorting_box = newcb(_("Sort files according to full path"),
+                              'fullpath_sorting')
```
Comment (ccordoba12) [inline spyder/plugins/editor.py]: Please remove this option. It's not need anymore in the 3.x branch.

File: spyder/plugins/editor.py [hunk 3]
```diff
@@ -105,9 +105,12 @@ def setup_page(self):
         
         interface_group = QGroupBox(_("Interface"))
         newcb = self.create_checkbox
+        fpsorting_box = newcb(_("Sort files according to full path"),
+                              'fullpath_sorting')
         showtabbar_box = newcb(_("Show tab bar"), 'show_tab_bar')
 
         interface_layout = QVBoxLayout()
+        interface_layout.addWidget(fpsorting_box)
```
Comment (ccordoba12) [inline spyder/plugins/editor.py]: Also remove this.

File: spyder/plugins/editor.py [hunk 4]
```diff
@@ -539,6 +542,9 @@ def restore_scrollbar_position(self):
     def get_plugin_title(self):
         """Return widget title"""
         title = _('Editor')
+        filename = self.get_current_filename()
+        if filename:
+            title += ' - '+to_text_string(filename)
```
Comment (ccordoba12) [inline spyder/plugins/editor.py]: All this new lines are not needed either.

File: spyder/plugins/editor.py [hunk 5]
```diff
@@ -834,6 +840,17 @@ def get_plugin_actions(self):
                    triggered=self.run_cell_and_advance,
                    context=Qt.WidgetShortcut)
 
+        re_run_last_cell_action = create_action(self,
+                   _("Re-run last cell"),
+                   icon=ima.icon('run_cell'),
```
Comment (ccordoba12) [inline spyder/plugins/editor.py]: Please leave this action without an icon.

File: spyder/plugins/editor.py [hunk 6]
```diff
@@ -1041,13 +1058,14 @@ def get_plugin_actions(self):
...
+                            run_cell_advance_action,
+                            re_run_last_cell_action, MENU_SEPARATOR,
                             run_selected_action, re_run_action,
                             configure_action, MENU_SEPARATOR]
         self.main.run_menu_actions += run_menu_actions
         run_toolbar_actions = [run_action, run_cell_action,
-                               run_cell_advance_action, run_selected_action,
-                               re_run_action]
+                               run_cell_advance_action,
+                               re_run_action, configure_action]
```
Comment (ccordoba12) [inline spyder/plugins/editor.py]: Please remove `configure_action` from here.

File: spyder/widgets/sourcecode/base.py [hunk 7]
```diff
@@ -621,18 +623,31 @@ def get_selection_as_executable_code(self):
         
         return ls.join(lines)
 
-    def get_cell_as_executable_code(self):
-        """Return cell contents as executable code"""
+    def __exec_cell(self):
+        init_cur = QTextCursor(self.textCursor())
```
Comment (ccordoba12) [inline spyder/widgets/sourcecode/base.py]: Rename this variable to `init_cursor`

**Comment by ccordoba12** [comment]
Thanks @andfoy!! I did a quick review to start with, but everything looks good to me.

A test is missing but I'll add it myself after you finish my review ;-)

File: spyder/plugins/editor.py [hunk 8]
```diff
@@ -1041,12 +1051,13 @@ def get_plugin_actions(self):
...
         run_menu_actions = [run_action, run_cell_action,
-                            run_cell_advance_action, MENU_SEPARATOR,
+                            run_cell_advance_action,
+                            re_run_last_cell_action, MENU_SEPARATOR,
                             run_selected_action, re_run_action,
                             configure_action, MENU_SEPARATOR]
         self.main.run_menu_actions += run_menu_actions
         run_toolbar_actions = [run_action, run_cell_action,
-                               run_cell_advance_action, run_selected_action,
+                               run_cell_advance_action,
```
Comment (ccordoba12) [inline spyder/plugins/editor.py]: Sorry for the confusion, please restore `run_selected_action` here. It was fine before :-)

**Pull request merged**